lm_eval/tasks/x_stance.py

```diff
@@ -0,0 +1,202 @@
+"""
+X-Stance: A Multilingual Multi-Target Dataset for Stance Detection
+http://ceur-ws.org/Vol-2624/paper9.pdf
+
+The x-stance dataset consists of more than 150 political questions and 67,000 comments by candidates.
+It can be used to train and evaluate stance detection systems.
+Comments are in German, French and Italian (for this task only the German subset has been selected). Questions are given in all three languages and English.
+The data have been extracted from the Swiss voting advice platform Smartvote.
+
+https://github.com/ZurichNLP/xstance
+"""
+import datasets
+from lm_eval.base import Task, rf
+from lm_eval.metrics import mean
+from functools import partial
+import numpy as np
+
+_CITATION = """
+@inproceedings{vamvas2020xstance,
+    author    = "Vamvas, Jannis and Sennrich, Rico",
+    title     = "{X-Stance}: A Multilingual Multi-Target Dataset for Stance Detection",
+    booktitle = "Proceedings of the 5th Swiss Text Analytics Conference (SwissText)  16th Conference on Natural Language Processing (KONVENS)",
+    address   = "Zurich, Switzerland",
+    year      = "2020",
+    month     = "jun",
+    url       = "http://ceur-ws.org/Vol-2624/paper9.pdf"
+}
+"""
+
+
+# Helper functions for aggregation (separate function for each metric)
+def _xstance_agg_precision(key, items):
+    references, predictions = zip(*items)
+    precision_metric = datasets.load_metric("precision")
+    return precision_metric.compute(
+        references=references,
+        predictions=predictions,
+        average="macro",
+        labels=np.unique(predictions),
+    )[key]
+
+
+def _xstance_agg_recall(key, items):
+    references, predictions = zip(*items)
+    recall_metric = datasets.load_metric("recall")
+    return recall_metric.compute(
+        references=references,
+        predictions=predictions,
+        average="macro",
+        labels=np.unique(predictions),
```

Comment:
np.unique may give different labels depending on the ordering of the predictions. Better use fixed labels like ["FAVOUR", "AGAINST"] (not sure about the correct order).